pydantic_ai_slim/pydantic_ai/models/instrumented.py

```diff
@@ -275,6 +277,10 @@ def messages_to_otel_events(messages: list[ModelMessage]) -> list[Event]:
                     **(event.attributes or {}),
                 }
             events.extend(message_events)
+        if last_model_request and last_model_request.instructions:
+            events.insert(
+                0, Event('gen_ai.system.message', body={'content': last_model_request.instructions, 'role': 'system'})
```

Comment:
Maybe setting the role to instructions would be clearer, but make sure it looks OK in the ui

Reply:
You mean for the UI?